metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -237,6 +265,94 @@ def create(cls, config_dict: Dict, ctx: PipelineContext) -> "SQLServerSource":
         config = SQLServerConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    def get_upstreams(self, inspector: Inspector, key: str) -> List[UpstreamClass]:
+        if not self.config.mssql_lineage:
+            return []
+        result = []
+        for dest in self.table_view_dependencies.get(key, []):
+            dest_name = self.get_identifier(
+                schema=dest.get("schema", ""),
+                entity=dest.get("name", ""),
+                inspector=inspector,
+            )
+            dest_urn = make_dataset_urn_with_platform_instance(
+                self.platform,
+                dest_name,
+                self.config.platform_instance,
+                self.config.env,
+            )
+            external_upstream_table = UpstreamClass(
+                dataset=dest_urn,
+                type=DatasetLineageTypeClass.COPY,
+            )
+            result.append(external_upstream_table)
+        return result
+
+    def _populate_stored_procedures_dependencies(self, conn: Connection) -> None:
+
+        path_to_stored_procedures_dependencies_query = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "sql_queries/stored_procedures_dependencies_query.sql",
+        )
+        path_to_stored_procedures_dependencies_select = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "sql_queries/stored_procedures_dependencies_select.sql",
+        )
+        stored_procedures_dependencies_query = open(
+            path_to_stored_procedures_dependencies_query, mode="r", encoding="utf-8"
+        ).read()
+        stored_procedures_dependencies_select = open(
+            path_to_stored_procedures_dependencies_select, mode="r", encoding="utf-8"
+        ).read()
+
+        trans = conn.begin()
+
+        conn.execute(stored_procedures_dependencies_query)
+
+        _dependencies = conn.execute(stored_procedures_dependencies_select)
+
+        for row in _dependencies:
+
+            if row:
+                _key = f"{row['current_db']}.{row['procedure_schema']}.{row['procedure_name']}"
+
+                self.procedures_dependencies[_key].append(
+                    {
+                        "referenced_database_name": row["referenced_database_name"],
+                        "referenced_schema_name": row["referenced_schema_name"],
+                        "referenced_entity_name": row["referenced_entity_name"],
+                        "referenced_object_type": row["referenced_object_type"],
+                        "is_selected": row["is_selected"],
+                        "is_select_all": row["is_select_all"],
+                        "is_updated": row["is_updated"],
+                    }
+                )
+
+        trans.commit()
```

Comment:
why do we need to commit this? shouldn't we be rolling it back?

also, do we need a try...finally to ensure rollback is called?